### PR TITLE
Replace ListView to FlatList

### PR DESCRIPTION
**Pull request by HichamELBSI**
Quick fix to https://github.com/tableflip/react-native-select-multiple/issues/15

### CODE REVIEW

File: src/SelectMultiple.js [hunk 1]
```diff
@@ -58,17 +58,12 @@ export default class SelectMultiple extends Component {
 
     const rows = this.getRowData(props)
 
-    const dataSource = new ListView.DataSource({
-      rowHasChanged: (r1, r2) => r1.value !== r2.value || r1.selected !== r2.selected
-    }).cloneWithRows(rows)
```
Comment (alanshaw) [inline src/SelectMultiple.js]: I'm not familiar with `FlatList`, is there no `rowHasChanged` equivalent?

File: src/SelectMultiple.js [hunk 2]
```diff
@@ -107,11 +102,18 @@ export default class SelectMultiple extends Component {
     }
   }
 
+  keyExtractor = (item, index) => index
```
Comment (alanshaw) [inline src/SelectMultiple.js]: Should this be a prop that can be passed in now?

File: src/SelectMultiple.js [hunk 3]
```diff
@@ -58,17 +58,12 @@ export default class SelectMultiple extends Component {
 
     const rows = this.getRowData(props)
 
-    const dataSource = new ListView.DataSource({
-      rowHasChanged: (r1, r2) => r1.value !== r2.value || r1.selected !== r2.selected
-    }).cloneWithRows(rows)
-
-    this.state = { dataSource }
+    this.state = { dataSource: rows }
```
Comment (alanshaw) [inline src/SelectMultiple.js]: Shall we just all this `data` now (since that's what `FlatList` calls it)?

File: src/SelectMultiple.js (hunk 1 above)
Comment (HichamELBSI) [reply src/SelectMultiple.js]: FlatList docs (https://facebook.github.io/react-native/docs/flatlist.html) uses PureComponent to handle optimization and avoid wasteful re-render 😅

**Comment by HichamELBSI** [comment]
i just push a new commit 😅

File: src/SelectMultiple.js [hunk 4]
```diff
@@ -19,18 +19,19 @@ const styleType = PropTypes.oneOfType([
...
     selectedItems: PropTypes.arrayOf(itemType),
 
     onSelectionsChange: PropTypes.func.isRequired,
+    keyExtractor: PropTypes.func,
 
     checkboxSource: sourceType,
     selectedCheckboxSource: sourceType,
     renderLabel: PropTypes.func,
-    listViewProps: PropTypes.any,
+    FlatListProps: PropTypes.any,
```
Comment (alanshaw) [inline src/SelectMultiple.js]: why capital F?

File: src/SelectMultiple.js [hunk 5]
```diff
@@ -107,11 +101,18 @@ export default class SelectMultiple extends Component {
...
-    const { style, listViewProps } = this.props
-    const { renderItemRow } = this
-    return <ListView style={style} dataSource={dataSource} renderRow={renderItemRow} {...(listViewProps || {})} />
+    const { style, flatListProps, items, keyExtractor } = this.props
+    return <FlatList
+        style={style}
+        keyExtractor={keyExtractor || this.keyExtractor}
+        data={dataSource}
+        renderItem={this.renderItemRow}
+        {...FlatListProps}
```
Comment (alanshaw) [inline src/SelectMultiple.js]: `FlatListProps` undefined here?

**Comment by ellman121** [comment]
@alanshaw Any updates on this?  Keep getting deprecation warning from React when using the library, would like to see it maintained.

**Comment by Kikketer** [comment]
Is this PR dead on the vine?  Running into the same issue that was originally reported.

**Comment by HichamELBSI** [comment]
@Kikketer Hi, I don't know why this branch is not merged. @alanshaw Are you still working on this package ?

**Pull request merged**